File: LiveTrackingSystem/LiveTracking.Api/Models/Holiday.cs
namespace LiveTracking.Api.Models;

public class Holiday
{
    public int HolidayId { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime Date { get; set; }
    public int Year { get; set; }
    public bool IsRecurring { get; set; } = false;
    public bool IsActive { get; set; } = true;
    public string? Description { get; set; }
}

File: LiveTrackingSystem/MM_Fuel_Apps/app/src/main/java/com/zynexbd/livetracking/models/Holiday.kt
package com.zynexbd.livetracking.models

import com.google.gson.annotations.SerializedName

data class Holiday(
    @SerializedName("holidayId")
    val holidayId: Int,
    @SerializedName("name")
    val name: String,
    @SerializedName("date")
    val date: String,
    @SerializedName("year")
    val year: Int,
    @SerializedName("isRecurring")
    val isRecurring: Boolean = false,
    @SerializedName("isActive")
    val isActive: Boolean = true,
    @SerializedName("description")
    val description: String? = null
)

data class CreateOrUpdateHolidayRequest(
    val name: String,
    val date: String,
    val isRecurring: Boolean = false,
    val isActive: Boolean = true,
    val description: String? = null
)
